Automattic/wp-calypso
# Focused Launch: Add focused launch modal from @automattic/launch package

## Changes proposed in this Pull Request

* Adds the Focused Launch Flow modal (as spec'd in p9Jlb4-1PL-p2)
* Changes are similar to #46646, but in this PR the code for the focused launch flow modal is inside the `@automattic/launch` package (as agreed in p1603116319449900-slack-C0KDTA48Y and following replies)
* This PR builds on top of #46516, but the modal is rendered inside the block editor's `iframe` — for this reason, the `EditorLaunchModal` component and its related code (including the `post-editor/launch-overlay` feature flag) have been deleted
* ~~The code for opening the Focused Launch Flow modal lies in Editing Toolkit, in the same place where the code for the current "Complete Setup" launch modal is (`apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/attach-launch-sidebar.tsx`)~~. **Edit**: moved this to [its own place](https://github.com/Automattic/wp-calypso/blob/add/focused-launch-modal/apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/attach-focused-launch.tsx). 

### Phab patches

- FSE: D51671-code
- WBE: D52152-code

### Sub tasks:
- [x] Avoid code duplication for `hooks` and `stores` folder (@alshakero , this PR)
- [x] Add "steps" HTML/CSS logic. Restructure HTML to allow for separate horizontal plans (@alshakero , this PR)
- [x] Add `gutenboarding/focused-launch-flow` flag to enable focused launch flow modal in alternative to control launch flow (@ciampo, this PR)
- [x] Refine overall typography styles (@alshakero #46784). 
- [x] Refine overall layout (e.g. the quotes, on mobile, should appear under each section title, @alshakero #46784)
- [x] register Launch data-store and any other necessary data-stores (registration example: https://github.com/Automattic/wp-calypso/blob/master/client/landing/gutenboarding/stores/domain-suggestions/index.ts) (@yansern / @ciampo https://github.com/Automattic/wp-calypso/pull/46834)

Remaining TODO items have been moved to separate issues in the [Focused Launch milestone](https://github.com/Automattic/wp-calypso/milestone/325)

## Testing instructions

Run the project locally. Make sure you have an open connection with your sandbox, and in 3 separate terminal windows:
   - `yarn start`
   - `cd apps/editing-toolkit && yarn dev --sync`
   - `cd apps/wpcom-block-editor && yarn dev --sync`

### The Focused Launch modal should open correctly in the block editor

1. Sandbox  both `widgets.wp.com` and an unlaunched site created **using `/start`**
2. Visit `http://calypso.localhost:3000/post/YOUR_UNLAUNCHED_SITE_CREATED_USING_START_HERE.wordpress.com/home?flags=create/focused-launch-flow,gutenboarding/persistent-launch-button`
3. Click on the "Launch" button in the header bar
4. Focused Launch modal should open

### The StepByStep Launch modal should open correctly in the block editor

1. Sandbox  both `widgets.wp.com` and an unlaunched site created **using `/new`**
2. Visit `http://calypso.localhost:3000/post/YOUR_UNLAUNCHED_SITE_CREATED_USING_NEW_HERE.wordpress.com/home`
3. Click on the "Launch" button in the header bar
4. StepByStep Launch modal should open


### The Focused Launch modal should open correctly in Calypso (used to speed up development)

1. Visit `http://calypso.localhost:3000/post/YOUR_SITE_HERE.wordpress.com/home?flags=create/focused-launch-flow-calypso`
2. It should load the focused launch modal (instead of the block editor)


### The block editor should function correctly when the `create/focused-launch-flow-calypso` is not enabled

1. Visit `http://calypso.localhost:3000/post/YOUR_SITE_HERE.wordpress.com/home`
2. It should load the block editor as usual.

### Smoke testing on Atomic

1. Create an atomic site (e.g ephemeral via Jurassic Ninja, and/or wp.com business plan with at least one plugin installed)
2. Install FSE as a plugin (see p1602862760430800-slack-C0KDTA48Y for more details)
3. Smoke test that nothing breaks when opening the block editor
4. Install Gutenberg plugin, repeat smoke test



Closes #46855
